File: sdk/android/sqljson/controller/HttpController.java
package ardvro.component.sqljson.controller;

import ardvro.core.connection.IConnection;
import ardvro.core.connection.IResponseListener;

public class HttpController
{
    private IConnection _conection;

    public HttpController(IConnection connection)
    {
        _conection = connection;
    }

    public void Get(String url, String authorization, IResponseListener onResponse)
    {
        String functionName = "ardvro/component/sqljson/Http/Get";
        _conection.Submit(new Object[]{ url, authorization }, functionName, onResponse);
    }

    public void Post(String url, Object data, String authorization, IResponseListener onResponse)
    {
        String functionName = "ardvro/component/sqljson/Http/Post";
        _conection.Submit(new Object[]{ url, data, authorization }, functionName, onResponse);
    }

    public void GetStream(String url, String authorization, IResponseListener<byte[]> onResponse)
    {
        String functionName = "ardvro/component/sqljson/Http/GetStream";
        _conection.Submit(new Object[]{ url, authorization }, functionName, onResponse);
    }

    public void PostStream(String url, Object data, String authorization, IResponseListener<byte[]> onResponse)
    {
        String functionName = "ardvro/component/sqljson/Http/PostStream";
        _conection.Submit(new Object[]{ url, data, authorization }, functionName, onResponse);
    }

    public void Upload(String url, Object data, byte[] bytes, String authorization, IResponseListener onResponse)
    {
        String functionName = "ardvro/component/sqljson/Http/Upload";
        _conection.Submit(new Object[]{ url, data, bytes, authorization }, functionName, onResponse);
    }

}

File: sdk/web/component/sqljson/controller/HttpController.js
var HttpController = function HttpController(cfg)
{
    let ctrl = {};

    ctrl.Get = function (url, authorization, onResponse)
    {
        let args = [url, authorization];
        cfg.Connector.Submit(args, "ardvro/component/sqljson/Http/Get", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.Post = function (url, data, authorization, onResponse)
    {
        let arg = {
            Url: url,
            Data: data,
            Authorization: authorization
        };
        cfg.Connector.Submit(arg, "ardvro/component/sqljson/Http/Post", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };
    
    ctrl.GetStream = function (url, authorization, onResponse)
    {
        let args = [url, authorization];
        cfg.Connector.Submit(args, "ardvro/component/sqljson/Http/GetStream", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.PostStream = function (url, data, authorization, onResponse)
    {
        let arg = {
            Url: url,
            Data: data,
            Authorization: authorization
        };
        cfg.Connector.Submit(arg, "ardvro/component/sqljson/Http/PostStream", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.Upload = function (url, data, bytes, authorization, onResponse)
    {
        let arg = {
            Url: url,
            Data: data,
            Bytes: bytes,
            Authorization: authorization
        };
        cfg.Connector.Submit(arg, "ardvro/component/sqljson/Http/Upload", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    return ctrl;
};